### PR TITLE
Hotfix

### PR DESCRIPTION
**Pull request by zmssp**
Hotfix for 3116,3118,3119,3120,3122

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.webapp/src/com/microsoft/azuretools/webapp/ui/AppServiceCreateDialog.java [hunk 1]
```diff
@@ -1022,17 +1022,19 @@ public static void initAspCache() {
...
-                    try {
-                        List<AppServicePlan> appServicePlans = AzureWebAppMvpModel.getInstance()
-                            .listAppServicePlanBySubscriptionId(sd.getSubscriptionId());
-                        map.put(sd.getSubscriptionId(), appServicePlans);
-                        subscriber.onCompleted();
-                    } catch (Exception e) {
-                        Exceptions.propagate(e);
-                    }
-                }).subscribeOn(Schedulers.io()), sdl.size()).subscribeOn(Schedulers.io()).toBlocking().subscribe();
+            if (sdl.size() > 0) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.webapp/src/com/microsoft/azuretools/webapp/ui/AppServiceCreateDialog.java]: Do we need to execute `subscriber.onCompleted();` when sdl.size()==0?

Comment (zmssp) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.webapp/src/com/microsoft/azuretools/webapp/ui/AppServiceCreateDialog.java]: Do not need, if the sdl.size == 0 , the code will go to subscribe....
image

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/AzureAnAction.java [hunk 2]
```diff
@@ -104,15 +104,19 @@ protected String getOperationName(AnActionEvent event) {
...
+                if (text.contains("spark") || text.contains("hdinsight")) {
+                    return TelemetryConstants.HDINSIGHT;
+                }
+                String place = event.getPlace().toLowerCase();
+                if (place.contains("spark") || place.contains("hdinsight")) {
+                    return TelemetryConstants.HDINSIGHT;
+                }
             }
+            return serviceName;
+        } catch (Exception ignore) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/AzureAnAction.java]: Should we do NPE verification or catch all exceptions?

Comment (zmssp) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/AzureAnAction.java]: I think we should just swallow all exception. Here is just for metrics, should not block our main process

**Pull request merged**